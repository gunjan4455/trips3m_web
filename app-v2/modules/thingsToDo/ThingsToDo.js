import React from 'react';
import { PropTypes } from 'prop-types';

import { getFlexObject } from 'helpers/utils';
import LoaderCard from 'modules/shared/loaders/LoaderCard';
import SliderWithTitle from 'modules/shared/SliderWithTitle';

class ThingsToDo extends React.Component {
  static propTypes = {
    data: PropTypes.object,
    layout: PropTypes.object,
    config: PropTypes.object,
    loaded: PropTypes.bool,
    errors: PropTypes.array,
    externalCss: PropTypes.string,
    trackSegmentEvent: PropTypes.func,
    fetchThingsToDoData: PropTypes.func
  };

  static defaultProps = {
    data: {},
    layout: {},
    config: {
      variant: 'FULL',
      order: 8
    },
    externalCss: '',
    loaded: false,
    trackSegmentEvent: () => {},
    fetchThingsToDoData: () => {}
  };

  componentDidMount() {
    this.fetchThingsToDoData(this.props);
  }

  componentDidUpdate(prevProps) {
    if (this.props.layout.metaInfo && prevProps.layout.metaInfo
      && this.props.layout.metaInfo.destination !== prevProps.layout.metaInfo.destination) {
      this.fetchThingsToDoData(this.props);
    }
  }

  fetchThingsToDoData = (props) => {
    const { config, layout, fetchThingsToDoData } = props;
    if (config && config.csr) {
      fetchThingsToDoData({
        destination: layout.metaInfo.destination,
        lookupParent: config.lookupParent
      });
    }
  };

  trackSegment = (cta, cardId, miscType1, miscId1) => {
    this.props.trackSegmentEvent({
      cta,
      object: cardId ? `Things to do/${cardId}` : 'Things to do',
      miscType1,
      miscId1
    });
  };

  render() {
    const { data, config, loaded, externalCss, errors } = this.props;
    if (errors && errors.length) {
      return null;
    }

    if (loaded && (!data.destinationSchemes || !data.destinationSchemes.length)) {
      return null;
    }

    return (
      <div className={externalCss}>
        <div className={`clearfix container pl0 pr0`}
             style={getFlexObject(config.variant, config.order)}>
          {
            !loaded
              ? <LoaderCard />
              : (
                data.destinationSchemes && data.destinationSchemes.length
                  ? <SliderWithTitle data={data} trackSegment={this.trackSegment} config={config} overflow='scroll' />
                  : null
              )
          }
        </div>
      </div>);
  }
}

export default ThingsToDo;
