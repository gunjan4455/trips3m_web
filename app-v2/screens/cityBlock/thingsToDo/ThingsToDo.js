import React, { Component } from 'react';
import PropTypes from 'prop-types';

import parseFilters from 'helpers/parseCityBlockFilters';
import Guide from 'modules/shared/Guide';
import CityBlockFilters from 'modules/cityBlock/CityBlockFilters';
import LoadMorePackages from 'modules/shared/loadMorePackages';
import NoPackageCard from 'modules/shared/NoPackageCard';
import ToVisitCard from './ToVisitCard';
import { showPopUp } from 'helpers/exitIntent';
import { useQuery } from "containers/Listing/parseFilters";

class ThingsToDo extends Component {
  constructor(){
    super();
    this.state= {
      currentPage: 1
    };
    this.resetFilters = this.resetFilters.bind(this);
  }

  resetFilters(){
    const values = {};
    this.props.resetFilters(values);
  }

  loadMoreThingsToDoCards = (destination, city, tab, experience, query) => {
    const { filters } = parseFilters({
      query: query || {}
    });
    const queryParams = {
      offset: this.state.currentPage === 1 ? (this.state.currentPage * 10) : (this.state.currentPage * 5),
      limit: 5,
      things_to_do: true,
      lookup_parent: false,
      city: city,
      tab: tab,
      experience: experience
    };
    const payload = { destination, filters, queryParams };
    this.props.loadMoreThingsToDoCards(payload);
    this.setState({currentPage: this.state.currentPage + 1});
  };

  render() {
    const { data, location, applyFilters, match: { params }, resetPath, route: { experience, indexTab }, exitIntent,
    history } = this.props;
    const { destination, city, tab } = params;

    return (
      <div className="clearfix">
        <div className="m8 mt0">
            <Guide
              guide={{ title: data.title, heading: data.heading, description: data.description }} />
        </div>
        <div id="reactExitIntentImage" className="mkt-mob-exit-intent relative pl8 pr8 mb8" onClick={showPopUp} />
        <div className="m8 mt0">
            <div className="col-xs-12 p0">
              {
                data.destinationSchemes && data.destinationSchemes.map((item, i) => (
                  <div className="mb8"><ToVisitCard item={item} key={i} /></div>
                ))
              }
            </div>
        </div>
        {
          data.destinationSchemes && data.destinationSchemes.length ?
            <div className="m8 mt0">
              <LoadMorePackages
                packagesRemaining={data.totalDestinationSchemes - data.destinationSchemes.length}
                loadMorePackages={() => this.loadMoreThingsToDoCards(destination, city, tab || indexTab, experience, useQuery(location.search))}
                currentPage={this.state.currentPage}
                title='remaining Things To Do)'
              />
            </div> :
            <div className="m8 mt0 p15 sbcw"><NoPackageCard destination={destination} /></div>
        }
        {
          data.filters && (data.filters.filterCategories || data.filters.filteredCities) ?
            <CityBlockFilters
              filters={data.filters}
              applyFilters={applyFilters}
              tab="things-to-do"
              params={params}
              enableReinitialize
              resetPath={resetPath}
              onSubmit={values => {
                applyFilters({ values, location, filters: data.filters, params, push: history.push });
              }}
              resetForm={this.resetFilters}
            />
            : null
        }
      </div>
    );
  }
}

ThingsToDo.propTypes = {
  data: PropTypes.object.isRequired,
  params: PropTypes.object.isRequired,
  location: PropTypes.object.isRequired,
  exitIntent: PropTypes.object.isRequired,
  applyFilters: PropTypes.func.isRequired,
  resetPath: PropTypes.func.isRequired,
  loadMoreThingsToDoCards: PropTypes.func,
  resetFilters: PropTypes.func.isRequired,
  match: PropTypes.object.isRequired,
  route: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired
};

ThingsToDo.defaultProps = {
  loadMoreThingsToDoCards: () => {},
  routes: []
};

export default ThingsToDo;
