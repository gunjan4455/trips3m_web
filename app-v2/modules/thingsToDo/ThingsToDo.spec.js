import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import ThingsToDo from './ThingsToDo';

configure({ adapter: new Adapter() });

const DEFAULT_PROPS = {
  thingsToDo: {
    destination_name: 'Karela',
    sub_heading: 'Your list of the best things to do, compiled on the basis of actual traveler experiences',
    view_all_url: '',
    activities: [
      {
        title: 'Lighthouse Beach in Kovalam',
        description: 'Lighthouse beach in Kerala is a beautiful sandy beach with a lighthouse. In a state that offers so much to travelers like dense forests, picturesque hill stations, lush green plantations, and serene backwaters, the hawa beach in Kovalam is a breath of fresh air.',
        image_url: 'https://www.hlimg.com/images/places2see/738X538/kerala-tours3.jpg',
        image_alt: 'https://www.hlimg.com/images/places2see/738X538/kerala-tours3.jpg',
        navigation_url: '',
        labels: ['Beaches', 'Nature'],
        url_spotlight: ''
      }]
  }
};

const shallowWrapperThingsToDo = (props) => {
  return shallow(<ThingsToDo {...props} />);
};

describe('ThingsToDo module', () => {
  let props = { ...DEFAULT_PROPS };

  afterEach(() => {
    props = { ...DEFAULT_PROPS };
  });

  it('should render correctly', () => {
    const wrapper = shallowWrapperThingsToDo(props);
    expect(wrapper).toMatchSnapshot();
  });
});
