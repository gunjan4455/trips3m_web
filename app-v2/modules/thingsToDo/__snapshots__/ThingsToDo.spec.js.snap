// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThingsToDo module should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ThingsToDo
    config={
      Object {
        "order": 8,
        "variant": "FULL",
      }
    }
    data={Object {}}
    externalCss=""
    fetchThingsToDoData={[Function]}
    layout={Object {}}
    loaded={false}
    thingsToDo={
      Object {
        "activities": Array [
          Object {
            "description": "Lighthouse beach in Kerala is a beautiful sandy beach with a lighthouse. In a state that offers so much to travelers like dense forests, picturesque hill stations, lush green plantations, and serene backwaters, the hawa beach in Kovalam is a breath of fresh air.",
            "image_alt": "https://www.hlimg.com/images/places2see/738X538/kerala-tours3.jpg",
            "image_url": "https://www.hlimg.com/images/places2see/738X538/kerala-tours3.jpg",
            "labels": Array [
              "Beaches",
              "Nature",
            ],
            "navigation_url": "",
            "title": "Lighthouse Beach in Kovalam",
            "url_spotlight": "",
          },
        ],
        "destination_name": "Karela",
        "sub_heading": "Your list of the best things to do, compiled on the basis of actual traveler experiences",
        "view_all_url": "",
      }
    }
    trackSegmentEvent={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="clearfix container pl0 pr0"
        style={
          Object {
            "flex": "",
            "marginBottom": "",
            "order": 8,
          }
        }
      >
        <LoaderCard
          cardVisible={Object {}}
          lineVisible={Object {}}
        />
      </div>,
      "className": "",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <LoaderCard
          cardVisible={Object {}}
          lineVisible={Object {}}
        />,
        "className": "clearfix container pl0 pr0",
        "style": Object {
          "flex": "",
          "marginBottom": "",
          "order": 8,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "cardVisible": Object {},
          "lineVisible": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="clearfix container pl0 pr0"
          style={
            Object {
              "flex": "",
              "marginBottom": "",
              "order": 8,
            }
          }
        >
          <LoaderCard
            cardVisible={Object {}}
            lineVisible={Object {}}
          />
        </div>,
        "className": "",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <LoaderCard
            cardVisible={Object {}}
            lineVisible={Object {}}
          />,
          "className": "clearfix container pl0 pr0",
          "style": Object {
            "flex": "",
            "marginBottom": "",
            "order": 8,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "cardVisible": Object {},
            "lineVisible": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
